src/Application.php

```diff
@@ -74,11 +75,20 @@ protected function initServices() {
         };
 
         $this['pdo'] = function() {
-            return new \PDO($this['config']['pdo']['dsn'], $this['config']['pdo']['username'], $this['config']['pdo']['password']);
+            $pdo = new PDO(
+                $this['config']['pdo']['dsn'],
+                $this['config']['pdo']['username'],
+                $this['config']['pdo']['password']
+            );
+            $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
```

Comment:
@dennisdegreef , well, utf8mb4 is something that needs to be set during table creation. Which we do. I think in case of mysql the current recommendation is to actually add `charset=utf8` to the mysql DSN, or run `SET NAMES 'UTF-8'`.